devauth/devauth.go

```diff
@@ -403,6 +404,15 @@ func (d *DevAuth) DecommissionDevice(ctx context.Context, devId string) error {
 }
 
 func (d *DevAuth) AcceptDeviceAuth(ctx context.Context, device_id string, auth_id string) error {
+	allow, err := d.canAcceptDevice(ctx)
```

Comment:
Here's the thing, in order to correctly apply the limit, we should first 'accept' the device and then verify if it's above the limit. Otherwise there's a race between checking agains the and updating a shared state in DB. IMO it will be possible to abuse it if you spam the status endpoint.

That's the theory. In practice, we've cut some corners already so this one might be acceptable too (even though it's wrong). Maybe at least leave a note in the code that this part is racy and we'll need to do something about this in the future.

Reply:
so, should the flow be like this?:
- mark auth set as accepted
- count
- if above limit - unmark?

that's 'better', but nontransactionality could still bite us. not sure if it's worth the effort, but I'll consider it.

Reply:
@bboozzoo added a note ;)